compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5617,10 +5617,393 @@ OMR::ARM64::TreeEvaluator::arraytranslateEvaluator(TR::Node *node, TR::CodeGener
+            generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, dstEndReg, dstReg, length);
+            generateMemSrc1Instruction(cg, op, node, TR::MemoryReference::createWithDisplacement(cg, dstReg, 0), vectorValueReg);
+            generateMemSrc1Instruction(cg, op, node, TR::MemoryReference::createWithDisplacement(cg, dstEndReg, offset), vectorValueReg);
+            srm->reclaimScratchRegister(dstEndReg);
+            }
+         }
+      else
+         {
+         const TR::InstOpCode::Mnemonic strOpCode = (!valueNode->getDataType().isFloatingPoint()) ?
+                                                      ((elementSize == 1) ? TR::InstOpCode::strbimm :
+                                                      (elementSize == 2) ? TR::InstOpCode::strhimm :
+                                                      (elementSize == 4) ? TR::InstOpCode::strimmw : TR::InstOpCode::strimmx) :
+                                                      valueNode->getDataType().isFloat() ? TR::InstOpCode::vstrimms : TR::InstOpCode::vstrimmd;
+         valueReg = isValueZero ? cg->allocateRegister() : cg->gprClobberEvaluate(valueNode);
+         for (int i = 0; i < length / elementSize; i++)
+            {
+            generateMemSrc1Instruction(cg, strOpCode, node, TR::MemoryReference::createWithDisplacement(cg, dstReg, i * elementSize), valueReg);
+            }
+         }
+      if ((valueReg != NULL) && isValueZero)
+         {
+         TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+         TR::RegisterDependencyConditions *conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 1, cg->trMemory());
+         conditions->addPostCondition(valueReg, TR::RealRegister::xzr);
+         generateLabelInstruction(cg, TR::InstOpCode::label, node, doneLabel, conditions);
+         }
+      }
+   else
+      {
+      /*
+       * Generating following instruction sequence
+       *
+       * cbz     lengthReg, LDONE
+       * dup     v0.16b, byteRegw
+       * add     endReg, dstReg, lengthReg
+       * cmp     lengthReg, #32
+       * b.cc    LessThan32
+       * stp     q0, q0, [dstReg]
+       * cmp     lengthReg, #96
+       * b.ls    LessThanOrEqual96
+       * sub     lengthReg, lengthReg, #96 ; 64 + 32
+       * ; align by 16
+       * and     remainderReg, dstReg, #15
+       * add     lengthReg, lengthReg, remainderReg
+       * and     dstReg, dstReg, #~15    ; dstReg points the nearest 16-byte boundary before the location that is read in the next load.
+       * mainLoop:
+       * subs    lengthReg, lengthReg, #64
+       * stp     q0, q0, [dstReg, #32]
+       * stp     q0, q0, [dstReg, #64]!
+       * b.hi    mainLoop
+       * stp     q0, q0, [dstEndReg, #-64]
+       * stp     q0, q0, [dstEndReg, #-32]
+       * b       LDONE
+       *
+       * LessThanOrEqual96:
+       * tbz     lengthReg, #6, LessThan64
+       * stp     q0, q0, [srcReg, #32]
```

Comment:
I think this is `dstReg` instead of `srcReg`.

Reply:
Fixed.